### PR TITLE
HEEDLS-626 Add variable result count to Centre Course Setup

### PR DESCRIPTION
**Pull request by AlexJacksonDS**
### JIRA link
https://softwiretech.atlassian.net/browse/HEEDLS-626

### Description
Add custom items per page to the Centre Course Setup page

### Screenshots
image
image

image
image

image
image

-----
### Developer checks
(Leave tasks unticked if they haven't been appropriate for your ticket.)

I have:
- [x] Run the formatter and made sure there are no IDE errors.
- [x] Written tests for the changes (accessibility tests, unit tests for controller, data services, services, view models, etc)
- [x] Manually tested my work with and without JavaScript. Full manual testing guidelines can be found here: https://softwiretech.atlassian.net/wiki/spaces/HEE/pages/6703648740/Testing
- [ ] Updated/added documentation in Swiki and/or Readme. Links (if any) are below:
  - [doc_1_here](link_1_here)
- [ ] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken.
- [x] Scanned over my own MR to ensure everything is as expected.

### CODE REVIEW

File: DigitalLearningSolutions.Web.Tests/ViewModels/TrackingSystem/CourseSetup/CourseSetupViewModelTests.cs [hunk 1]
```diff
@@ -103,12 +105,39 @@ public void CourseSetupViewModel_filters_should_be_set()
...
+                null
             );
 
             // Then
             model.Filters.Should().BeEquivalentTo(expectedFilters);
             model.Courses.First().CourseName.Should().BeEquivalentTo("K");
         }
+
+        [Test]
+        public void CourseSetupViewModel_with_custom_items_per_page_should_to_return_the_specified_number_of_delegates()
```
Comment (livzorn) [inline DigitalLearningSolutions.Web.Tests/ViewModels/TrackingSystem/CourseSetup/CourseSetupViewModelTests.cs:118]: I think a small typo in this test name

**Review by livzorn — APPROVED** [review]
Just one small nitpick, otherwise looks good! No need for rereview

File: DigitalLearningSolutions.Web.Tests/ViewModels/TrackingSystem/CourseSetup/CourseSetupViewModelTests.cs (hunk 1 above)
Comment (livzorn) [reply DigitalLearningSolutions.Web.Tests/ViewModels/TrackingSystem/CourseSetup/CourseSetupViewModelTests.cs:118]: Oh wait maybe "should to" is a British thing?

File: DigitalLearningSolutions.Web/ViewModels/TrackingSystem/CourseSetup/CourseSetupViewModel.cs [hunk 2]
```diff
@@ -16,8 +16,9 @@ public class CourseSetupViewModel : BaseSearchablePageViewModel
             string sortBy,
             string sortDirection,
             string? filterBy,
-            int page
-        ) : base(searchString, page, true, sortBy, sortDirection, filterBy, searchLabel: "Search courses")
+            int page,
+            int? itemsPerPage
+        ) : base(searchString, page, true, sortBy, sortDirection, filterBy, itemsPerPage ?? DefaultItemsPerPage, searchLabel: "Search courses")
```
Comment (DanBloxham-sw) [inline DigitalLearningSolutions.Web/ViewModels/TrackingSystem/CourseSetup/CourseSetupViewModel.cs:21]: Minor, but the "searchLabel: " key is now redundant and can be safely removed.

**Review by DanBloxham-sw — CHANGES_REQUESTED** [review]
Very minor clean up change.

**Review by DanBloxham-sw — APPROVED** [review]

**Pull request merged**